### DISCUSSION TITLE
How to embed Dashboard filters into HTML

### DISCUSSION THREAD

**Discussion by aimmon** [apache/superset#25270] [Q&A / Help] [answered]
The current implementation feature is to embed Dashboard into iframe, but now I want to add filters as well

![da111](https://github.com/apache/superset/assets/20752671/3bb4d32b-cfcd-4323-935e-fe807193faa9)

But now I want to add filters as well
![9ECD2114-B8B6-4431-B3FC-530789044E86](https://github.com/apache/superset/assets/20752671/54b688e2-0815-4d75-80fa-f0278e34cdd5)

How to achieve

**Comment by sfirke** [comment, marked as answer]
You should be able to embed a dashboard with its filters in an iframe.  Do you have an argument like `show_filters=0` in your iframe URL?  That would hide the filters, which is what I normally do to embed a dashboard without the filter pane but that's something you can control.  See also `expand_filters`, both documented here: https://superset.apache.org/docs/creating-charts-dashboards/creating-your-first-dashboard/.

I'm going to convert this into a discussion, as it's not a bug report.

**Reply by aimmon** [reply]
thank you very much,It's working

**Reply by vikashrajgupta** [reply]
@sfirke - Is it possible to get the applied filters state in the embedded dashboard? I have embedded a superset dashboard in Ifram and it's working perfectly file I can apply the filters as well.
I just wanted to pick all of the filters whenever user will apply and click on a specific button.

**Reply by sfirke** [reply]
I don't think I follow, sorry.  Do you want the user to see the filter pane?

**Reply by vikashrajgupta** [reply]
@sfirke  No, the Filter pane is visible, I want to get the filter state in the embedded dashboard.
For example - When a user applies 2-3 filters from the filter pane and it clicks on apply filters after that I can get the applied filter values in Iframe.
Like here is a get request `api/v1/dashboard/{pk}/filter_state` that provides the filter values but not for embedded Iframe

**Reply by vikashrajgupta** [reply]
Here is the  [issue](https://github.com/apache/superset/discussions/25609) link for more explanation

**Reply by sfirke** [reply]
Ah okay that's not a part of Superset I'm familiar with. Maybe try the #embedding channel on the Superset Slack chat?

**Reply by vikashrajgupta** [reply]
Sure will check it out @sfirke
